Adding SeqKit 2.3.1 as part of Docker Bioinformatics Workshop

hkunerth opened the pull request:
<!--
Thank your for contributing to the Staph-B community!
Please fill in the appropriate checklist below and delete whatever is not relevant.

These recommendations are not meant as a means to keep people from contributing. 
If you need assistance with the following measures, submit your pull request anyway, and we will work with you...
Or you can contact us beforehand via slack or by sumitting a pull request https://github.com/StaPH-B/docker-builds/issues

Documentation on how to contribute can be found at https://staphb.org/docker-builds/contribute/
Documentation for how to create a Dockerfile can be found at https://staph-b.github.io/docker-builds/make_containers
A recommended Dockerfile template can be found at https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile


Please replace all '[ ]' with '[X]' to demonstrate completion.
-->

I have written and tested a Dockerfile to run SeqKit 2.3.1 and added it to my fork of the docker-builds repository. I have edited the readme and licenses markdown files with the appropriate information. I have also added a simple README.md file to the /seqkit/2.3.1 directory.

Pull Request (PR) checklist:
- [X ] Include a description of what is in this pull request in this message.
- [ X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [ X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [ X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [ ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [ X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [ X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [ X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing


<!-- If this PR is for something else, please add extra descriptions -->

erinyoung commented on README.md:
```diff
@@ -131,9 +131,10 @@ To learn more about the docker pull rate limits and the open source software pro
 | [SeqSero](https://hub.docker.com/r/staphb/seqsero/) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/seqsero)](https://hub.docker.com/r/staphb/seqsero) | <ul><li>1.0.1</li></ul> | https://github.com/denglab/SeqSero |
 | [SeqSero2](https://hub.docker.com/r/staphb/seqsero2/) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/seqsero2)](https://hub.docker.com/r/staphb/seqsero2) | <ul><li>0.1.0</li><li>1.0.0</li><li>1.0.2</li><li>1.1.0</li><li>1.1.1</li><li>1.2.1</li></ul> | https://github.com/denglab/SeqSero2/ |
-| [seqtk](https://hub.docker.com/r/staphb/seqtk) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/seqtk)](https://hub.docker.com/r/staphb/seqtk) | <ul><li>1.3</li></ul> | https://github.com/lh3/seqtk |
+| [seqtk](https://hub.docker.com/r/staphb/seqtk) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/seqtk)](https://hub.docker.com/r/staphb/seqtk) | <ul><li>2.3.1</li></ul> | https://github.com/lh3/seqtk |
```
Could you change seqtk's version back to 1.3?

erinyoung commented on seqkit/2.3.1/Dockerfile:
```diff
@@ -0,0 +1,47 @@
+LABEL base.image="ubuntu:xenial"
+LABEL dockerfile.version="1"
+LABEL software="SeqKit"
+LABEL software.version=$SOFTWARENAME_VER
```
These LABELs look great! Can you change software.version to 

```
LABEL software.version=$SEQKIT_VER
```

erinyoung commented on seqkit/2.3.1/Dockerfile:
```diff
@@ -0,0 +1,47 @@
+LABEL maintainer.email="henrykunerth@gmail.com"
+
+# Install dependences (update as needed)
+RUN apt-get update && apt-get install -y \
+	wget
+
```
I love a tool that doesn't have a ton of dependencies! 

I do have a request to adjust this, however, because apt-get stores a lot of things, so your image will bigger than it needs to be.

Could you adjust this to something like
```
RUN apt-get update && apt-get install -y --no-install-recommends \
 wget && \
 apt-get autoclean && rm -rf /var/lib/apt/lists/*
```

erinyoung commented on seqkit/2.3.1/README.md:
```diff
@@ -0,0 +1,18 @@
+[**Tutorial**](http://bioinf.shenwei.me/seqkit/tutorial/),
+and 
+[**Benchmark**](http://bioinf.shenwei.me/seqkit/benchmark/))
+
```
Could you include a sample usage command?

Something like 
```
seqkit stat sample.fasta
```

erinyoung commented:
This looks great!

Yes, the tests failed because you're missing a value LABEL

```
FATAL : software.version label not found in seqkit:2.3.1
Please see template for recommended format https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile
```

But! The test in your Dockerfile worked great
```
#10 [test 1/1] RUN wget -P /data https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa && 	chmod 755 SRR13957123.consensus.fa && 	seqkit stat *
#0 0.053 --2023-01-27 18:43:49--  https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa
#0 0.057 Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.110.133, ...
#0 0.061 Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
#0 0.068 HTTP request sent, awaiting response... 200 OK
#10 0.232 Length: 29837 (29K) [text/plain]
#10 0.233 Saving to: '/data/SRR13957123.consensus.fa'
#10 0.233 
#10 0.233      0K .......... .......... .........                       100% 30.1M=0.001s
#10 0.234 
#10 0.235 2023-01-27 18:43:50 (30.1 MB/s) - '/data/SRR13957123.consensus.fa' saved [29837/29837]
#10 0.235 
#10 0.270 file                      format  type  num_seqs  sum_len  min_len  avg_len  max_len
#10 0.270 SRR13957123.consensus.fa  FASTA   DNA          1   29,778   29,778   29,778   29,778
#10 DONE 0.3s
```

erinyoung commented:
List of requested changes:

- Fix LABEL software.version
- reduce apt-get size
- add example usage in tool-specific readme

hkunerth commented:
Awesome! Thanks for the very helpful feedback. I'm learning a lot. 

I've updated the apt-get section (and added a ca-certificates install since it broke without it after the suggested change), added example usage to the readme, fixed the version number typo for seqtk (oops), and corrected the software.version label to include the correct argument. 

I think that should be everything, but I'll run one more github actions test to be certain.

hkunerth commented:
> I think that should be everything, but I'll run one more github actions test to be certain.

Passed. Thanks again!

erinyoung reviewed (approved):
Looks good to me!

Your test works great
```
#10 [test 1/1] RUN wget -P /data https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa && 	chmod 755 SRR13957123.consensus.fa && 	seqkit stat *
#0 0.059 --2023-01-27 22:01:25--  https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa
#0 0.061 Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.108.133, ...
#0 0.063 Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
#0 0.090 HTTP request sent, awaiting response... 200 OK
#10 0.298 Length: 29837 (29K) [text/plain]
#10 0.298 Saving to: '/data/SRR13957123.consensus.fa'
#10 0.299 
#10 0.299      0K .......... .......... .........                       100% 18.6M=0.002s
#10 0.301 
#10 0.301 2023-01-27 22:01:26 (18.6 MB/s) - '/data/SRR13957123.consensus.fa' saved [29837/29837]
#10 0.301 
#10 0.325 file                      format  type  num_seqs  sum_len  min_len  avg_len  max_len
#10 0.325 SRR13957123.consensus.fa  FASTA   DNA          1   29,778   29,778   29,778   29,778
#10 DONE 0.3s
```

I'm going to merge this and push this image to Dockerhub and quay. It _should_ be available within the hour.

hkunerth commented:
Great! Thanks for all your help with this, here and during the workshop!

Henry

On Fri, Jan 27, 2023 at 4:05 PM Young ***@***.***> wrote:

> Merged #576 <https://github.com/StaPH-B/docker-builds/pull/576> into
> master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StaPH-B/docker-builds/pull/576#event-8377460418>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC4UPHW43OMKWJHGK3LJ4Q3WURBCTANCNFSM6AAAAAAUI6TEJQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
